coldfront/plugins/qumulo/static/allocation.js

```diff
@@ -22,30 +22,38 @@ let confirmed = false;
 
 const submitButton = document.getElementById("allocation_form_submit");
 submitButton.addEventListener("click", (event) => {
-  const id_project_pk_elem = document.getElementById("div_id_project_pk");
-  const smb = protocols.find((protocol) => protocol.value === "smb");
+  const smb = protocols.find((protocol) => protocol.value === "smb"),
+        modalShouldDisplay = () => {
+          /*
+           * bmulligan (20241114): we are now using the document title to
+           * determine whether we are on the "Create Allocation" form.
+          */
+          if( document.title.trim() !== 'Create Allocation' )
+            return false;
 
-  // NOTE: we're using id_project_pk to determine whether we are on a
-  // parent or sub-allocation creation page
-  if (id_project_pk_elem && !smb.checked && !confirmed) {
```

Comment:
`id_project_pk_elem` was part of this check to prevent the SMB modal from appearing for sub-allocations, since those shouldn't have SMB shares defined by default.

Reply:
the trouble is that in the current `main` branch, that value gets populated by this call:

  `document.getElementById("div_id_project_pk");`
  
  ...and that call is the only occurrence of the string "div_id_project_pk" in the entire repo (branch: main), so there is no element with that ID in any of the HTML content.

Reply:
This element is generated by Django as part of it's templating language

Reply:
But as I see it this logic is just moved to line 36, correct?

```
 else if( document.getElementById("div_id_project_pk") === null )
            return false;
```

I'm not sure what the question is here, @peaches7083 ?